examples/hybrid/sphere/deformation_flow.jl

```diff
@@ -1,434 +1,335 @@
-    @. dρq3 -= alpha * vdivf2c(Ic2f(uₕ * ρq3))
-    # 2) Horizontal transport for ρq3 (includes both horizontal and vertical velocity components)
-    @. dρq3 -= alpha * hdiv(cuvw * ρq3) - alpha * ystar.ρq3
-
-    # 1) Vertical transport for ρq4:
-    # 1.1) vertical advection by vertical velocity, corrected by BB FCT:
-    @. ρq4_td =
-        beta * ρq4 -
-        alpha * vdivf2c(Ic2f(ρ) * first_order_upwind_c2f(w, ρq4 / ρ))
-    @. dρq4 =
-        beta * (dρq4 - ρq4) + ρq4_td -
-        alpha * vdivf2c(
-            Ic2f(ρ) * FCTBB(
-                (
-                    third_order_upwind_c2f(w, ρq4 / ρ) -
-                    first_order_upwind_c2f(w, ρq4 / ρ)
-                ),
-                ρq4_td / ρ / alpha,
-            ),
-        )
+function conservation_errors(sol)
+    initial_total_mass = sum(sol.u[1].c.ρ)
+    initial_tracer_masses = map(n -> sum(sol.u[1].c.ρq.:($n)), 1:5)
+    final_total_mass = sum(sol.u[end].c.ρ)
+    final_tracer_masses = map(n -> sum(sol.u[end].c.ρq.:($n)), 1:5)
+    return (
+        (final_total_mass - initial_total_mass) / initial_total_mass,
+        (final_tracer_masses .- initial_tracer_masses) ./ initial_tracer_masses,
+    )
+end
+tracer_roughnesses(sol) =
+    map(1:5) do n
+        q_n = sol.u[end].c.ρq.:($n) ./ sol.u[end].c.ρ
+        mean_q_n = mean(q_n) # TODO: replace the mean with a low-pass filter
+        return mean(abs.(q_n .- mean_q_n))
+    end
+tracer_ranges(sol) =
+    map(1:5) do n
+        q_n = sol.u[end].c.ρq.:($n) ./ sol.u[end].c.ρ
+        return maximum(q_n) - minimum(q_n)
+    end
 
-    # 1.2) vertical advection by horizontal velocity:
-    @. dρq4 -= alpha * vdivf2c(Ic2f(uₕ * ρq4))
-    # 2) Horizontal transport for ρq4 (includes both horizontal and vertical velocity components)
-    @. dρq4 -= alpha * hdiv(cuvw * ρq4) - alpha * ystar.ρq4
+ref_sol = run_deformation_flow(false, nothing)
+lim_sol = run_deformation_flow(true, nothing)
+fct_sol = run_deformation_flow(false, FCTZalesak)
 
-    Spaces.weighted_dss!(dydt.ρ)
-    Spaces.weighted_dss!(dydt.ρq1)
-    Spaces.weighted_dss!(dydt.ρq2)
-    Spaces.weighted_dss!(dydt.ρq3)
-    Spaces.weighted_dss!(dydt.ρq4)
+ref_ρ_err, ref_ρq_errs = conservation_errors(ref_sol)
+lim_ρ_err, lim_ρq_errs = conservation_errors(lim_sol)
+fct_ρ_err, fct_ρq_errs = conservation_errors(fct_sol)
 
-    return dydt
-end
+# Check that the conservation errors are not too big.
+@test abs(ref_ρ_err) < (FT == Float64 ? 1e-14 : 2e-5)
+@test all(abs.(ref_ρq_errs) .< 3 * abs(ref_ρ_err))
 
-# Set up RHS function
-ystar = copy(y0)
-y_td = copy(y0)
+# Check that the limiter and FCT have no effect on ρ.
+@test ref_ρ_err == lim_ρ_err == fct_ρ_err
 
-parameters = (; τ, coords, face_coords, ystar, y_td)
+# Check that the error of the tracer with q = 1 is nearly the same as that of ρ.
+@test ref_ρq_errs[5] ≈ ref_ρ_err atol = 2eps(FT)
+@test lim_ρq_errs[5] ≈ lim_ρ_err atol = eps(FT)
+@test fct_ρq_errs[5] ≈ fct_ρ_err atol = 2eps(FT)
```

Comment:
I changed these checks to use `atol` based on this [discourse](https://discourse.julialang.org/t/approximate-equality/8952/38). The two numbers were really really close and really really small (both of the same order of machine epsilon)